cave/utils/csv2rh.py

```diff
@@ -0,0 +1,250 @@
+__license__ = "3-clause BSD"
+__maintainer__ = "Joshua Marben"
+__email__ = "joshua.marben@neptun.uni-freiburg.de"
+
+class CSV2RH(object):
+    def __init__(self):
+        pass
+
+    def read_csv_to_rh(self, data,
+                       pcs:Union[None, str, ConfigurationSpace]=None,
+                       configurations:Union[None, str, dict]=None,
+                       train_inst:Union[None, str, list]=None,
+                       test_inst:Union[None, str, list]=None,
+                       instance_features:Union[None, str, dict]=None,
+                       logger=None,
+                       seed=42):
+        """ Interpreting a .csv-file as runhistory.
+        Expecting file with headers:
+            p_parameter1, p_parameter2, ...,
+            i_instancefeature1 (o), i_instancefeature2 (o), ...,
+            seed (o), cost, time (o), status (o)
+
+        Parameters:
+        -----------
+        csv_path: str
+            relative path to csv-file from executing directory
+        pcs: Union
+
+        Returns:
+        --------
+        rh: RunHistory
+            runhistory with all the runs from the csv-file
+        """
+        self.logger = logging.getLogger('cave.utils.csv2rh')
+        self.input_reader = InputReader()
+        self.pcs = pcs
+        self.train_inst = input_reader.read_instance_file(train_inst) if type(train_inst) == str else train_inst
+        self.test_inst =  input_reader.read_instance_file(test_inst) if type(test_inst) == str else test_inst
+        self.instance_features = input_reader.read_instance_features_file(instance_features) if type(instance_features) == str else instance_features
+        self.id_to_config = configurations if configurations else {}
+        self.cs = None
+
+        # Read in data
+        if isinstance(data, str):
+            csv_path = data
+            with open(csv_path, 'r') as csv_file:
+                csv_data = list(csv.reader(csv_file, delimiter=',',
+                                           skipinitialspace=True))
+            header, csv_data = csv_data[0], np.array([csv_data[1:]])[0]
+            self.data = pd.DataFrame(csv_data, columns=header)
+            self.data = self.data.apply(pd.to_numeric, errors='ignore')
+            self.logger.debug("Headers: " + str(list(self.data.columns.values)))
+        else:
+            self.data = data
+
+        # Expecting header as described in docstring
+        valid_values = ['seed', 'cost', 'time', 'status', 'config_id', 'instance_id']
+        if not len(self.data.columns) == len(set(self.data.columns)):
+            raise ValueError("Detected a duplicate in the columns of the "
+                             "csv-file \"%s\"." % csv_path)
+        for column_name in self.data.columns:
+            if not (column_name in valid_values or
+                    column_name.startswith('p_') or  # parameter
+                    column_name.startswith('i_')):  # instance feature
```

Comment:
This is a good start. However marking parameter and instance-features via p_ and i_ is very limiting. I'd suggest to use an input flag that says how many parameter values there are. Then the first X columns that are not in "valid_values" are parameters and all other values are instance features. If the number of parameters was not specified, we can treat all columns that are not in "valid_values" as parameters. This should make this interface more general and less error prone.